msgraphforentra_view.py

```diff
@@ -0,0 +1,89 @@
+# File: msgraphforentra_view.py
+#
+# Copyright (c) 2022-2024 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific language governing permissions
+# and limitations under the License.
+#
+
+
+def get_ctx_result(result):
+
+    ctx_result = {}
+    param = result.get_param()
+    summary = result.get_summary()
+    data = result.get_data()
+
+    ctx_result["param"] = param
+
+    if data:
+        ctx_result["data"] = data
+
+    if summary:
+        ctx_result["summary"] = summary
+
+    return ctx_result
+
+
+def display_risk_detections(provides, all_app_runs, context):
+
+    context["results"] = results = []
+
+    for summary, action_results in all_app_runs:
+        for result in action_results:
+            ctx_result = get_ctx_result(result)
+            if not ctx_result:
+                continue
+            results.append(ctx_result)
+
+    return "microsoftentra_list_risk_detections.html"
+
+
+def display_risky_users(provides, all_app_runs, context):
+
+    context["results"] = results = []
+
+    for summary, action_results in all_app_runs:
+        for result in action_results:
+            ctx_result = get_ctx_result(result)
+            if not ctx_result:
+                continue
+            results.append(ctx_result)
+
+    return "microsoftentra_list_risky_users.html"
+
+
+def display_alerts(provides, all_app_runs, context):
+
+    context["results"] = results = []
+
+    for summary, action_results in all_app_runs:
+        for result in action_results:
+            ctx_result = get_ctx_result(result)
+            if not ctx_result:
+                continue
+            results.append(ctx_result)
+
+    return "microsoft365defender_list_alerts.html"
+
+
+def update_alert(provides, all_app_runs, context):
+
+    context["results"] = results = []
+
+    for summary, action_results in all_app_runs:
+        for result in action_results:
+            ctx_result = get_ctx_result(result)
+            if not ctx_result:
+                continue
+            results.append(ctx_result)
+
+    return "microsoft365defender_update_alert.html"
```

Comment:
These functions are all doing the same thing, this is a great opportunity to dedupe code:

```suggestion
def add_results_to_context(all_app_runs, context):
    context["results"] = results = []

    for summary, action_results in all_app_runs:
        for result in action_results:
            if (ctx_result := get_ctx_result(result)):
                results.append(ctx_result)

def display_risk_detections(provides, all_app_runs, context):
    add_results_to_context(all_app_runs, context)
    return "microsoftentra_list_risk_detections.html"


def display_risky_users(provides, all_app_runs, context):
    add_results_to_context(all_app_runs, context)
    return "microsoftentra_list_risky_users.html"


def display_alerts(provides, all_app_runs, context):
    add_results_to_context(all_app_runs, context)
    return "microsoft365defender_list_alerts.html"


def update_alert(provides, all_app_runs, context):
    add_results_to_context(all_app_runs, context)
    return "microsoft365defender_update_alert.html"
```

Reply:
Perfect, agreed!